src/plugins/interactive_setup/public/manual_configuration_form.tsx

```diff
@@ -0,0 +1,436 @@
+  onSuccess?(): void;
+}
+
+export const ManualConfigurationForm: FunctionComponent<ManualConfigurationFormProps> = ({
+  defaultValues = {
+    host: 'https://localhost:9200',
+    username: 'kibana_system',
+    password: '',
+    caCert: '',
+  },
+  onCancel,
+  onSuccess,
+}) => {
+  const http = useHttp();
+
+  const [state, pingCluster] = useAsyncFn(async (values: ManualConfigurationFormValues) => {
+    const requiresCert = values.host.trim().startsWith('https://');
+    let requiresAuth = false;
+    let body: PingResponse;
+
+    try {
+      const response = await http.post<PingResponse>('/internal/interactive_setup/ping', {
+        body: JSON.stringify({
+          hosts: [values.host],
+        }),
+        asResponse: true,
+      });
+      body = response.body!;
+    } catch (error) {
+      if (error.response?.status !== 401) {
+        throw error;
+      }
+      requiresAuth = true;
+      body = error.body;
+    }
+
+    return { requiresCert, requiresAuth, ...body };
+  });
+
+  const [form, eventHandlers] = useForm({
+    defaultValues,
+    validate: async (values) => {
+      const errors: ValidationErrors<typeof values> = {};
+
+      let requiresAuth = false;
+      let requiresCert = false;
+      let needsToUploadCert = false;
+
+      if (!values.host) {
+        errors.host = i18n.translate('interactiveSetup.manualConfigurationForm.hostRequiredError', {
+          defaultMessage: 'Enter an address.',
+        });
+      } else {
+        try {
+          const url = new URL(values.host);
+          if (!url.protocol || !url.hostname || !url.port) {
+            throw new Error();
+          }
+          if (url.protocol === 'https:') {
+            requiresCert = true;
+          }
+        } catch (error) {
+          errors.host = i18n.translate(
+            'interactiveSetup.manualConfigurationForm.hostInvalidError',
+            {
+              defaultMessage: 'Enter a valid address including protocol and port number.',
+            }
+          );
+        }
+      }
+
+      if (!errors.host) {
+        try {
+          const data = await pingCluster(values);
+          if (data instanceof Error) {
+            throw data;
+          }
+          const { statusCode, certificateChain } = data;
+          requiresAuth = statusCode === 401;
+          needsToUploadCert = !certificateChain?.length;
+        } catch (error) {
+          errors.host = i18n.translate(
+            'interactiveSetup.manualConfigurationForm.hostConnectionError',
+            {
+              defaultMessage: "Can't connect to this cluster.",
+            }
+          );
+        }
+      }
+
+      if (requiresAuth) {
+        if (!values.username) {
+          errors.username = i18n.translate(
+            'interactiveSetup.manualConfigurationForm.usernameRequiredError',
+            {
+              defaultMessage: 'Enter a username.',
+            }
+          );
+        } else if (values.username === 'elastic') {
```

Comment:
__nit:__ we'll have to do this check on the server-side anyway, so maybe it'd make sense to move it there right from the start?

Reply:
Good point, this needs to be added to server side as well. 

It's best to give users immediate feedback though so they don't fill in the rest of the form unnecessarily with invalid credentials so will keep in both places.